File: src/main/java/com/github/xumeng/java/problems/easy/problem0234/Solution0234.java
package com.github.xumeng.java.problems.easy.problem0234;

import com.github.xumeng.java.common.datastructure.ListNode;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Solution0234 {

    public boolean isPalindrome(ListNode head) {
        List<Integer> elementList = new ArrayList<>();
        ListNode node = head;
        do {
            elementList.add(node.value);
            node = node.next;
        } while (node != null);
        int size = elementList.size();
        for (int i = 0, j = size - 1; i <= j; i++, j--) {
            Integer e1 = elementList.get(i);
            Integer e2 = elementList.get(j);
            if (!Objects.equals(e1, e2)) {
                return false;
            }
        }
        return true;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0234/Solution0234.kt
package com.github.xumeng.kotlin.problems.easy.problem0234

import com.github.xumeng.kotlin.common.datastructure.ListNode

class Solution0234 {

    fun isPalindrome(head: ListNode?): Boolean {
        val elementList = mutableListOf<Int>()
        var node = head
        do {
            if (node != null) {
                elementList.add(node.value)
                node = node.next
            }
        } while (node != null)
        var i = 0
        var j = elementList.size - 1
        while (i <= j) {
            val e1: Int = elementList.get(i)
            val e2: Int = elementList.get(j)
            if (e1 != e2) {
                return false
            }
            i++
            j--
        }
        return true
    }

}